### PR TITLE
Benchmark improvements

### PR DESCRIPTION
**Pull request by jamesdbrock**
Added and removed some benchmarks.
Benchmark section headers.

### CODE REVIEW

**Comment by jamesdbrock** [comment]
<table><tr><th><h2>digit 10000</h2></th></tr><tr><td><b>
runParser many digit 10000
</b>
<pre>
mean   = 13.63 ms
stddev = 8.96 ms
min    = 8.54 ms
max    = 68.19 ms
</pre></td></tr><tr><td><b>
runParser manyRec digit 10000
</b>
<pre>
mean   = 6.16 ms
stddev = 2.35 ms
min    = 4.91 ms
max    = 16.94 ms
</pre></td></tr><tr><td><b>
runParser Array.many digit 10000
</b>
<pre>
mean   = 71.82 ms
stddev = 12.54 ms
min    = 61.59 ms
max    = 110.64 ms
</pre></td></tr><tr><td><b>
StringParser manyRec CodePoints.anyDigit 10000
</b>
<pre>
mean   = 6.53 ms
stddev = 4.24 ms
min    = 4.50 ms
max    = 19.47 ms
</pre></td></tr><tr><td><b>
StringParser manyRec CodeUnits.anyDigit 10000
</b>
<pre>
mean   = 4.89 ms
stddev = 609.48 μs
min    = 4.43 ms
max    = 10.88 ms
</pre></td></tr><tr><td><b>
Regex.match \d* 10000
</b>
<pre>
mean   = 318.22 μs
stddev = 122.13 μs
min    = 281.80 μs
max    = 1.58 ms
</pre></td></tr><tr><th><h2>string 100000</h2></th></tr><tr><td><b>
runParser many string
</b>
<pre>
mean   = 26.11 ms
stddev = 7.60 ms
min    = 21.70 ms
max    = 61.61 ms
</pre></td></tr><tr><td><b>
runParser manyRec string
</b>
<pre>
mean   = 16.68 ms
stddev = 6.99 ms
min    = 13.06 ms
max    = 52.06 ms
</pre></td></tr><tr><td><b>
Regex.match literal*
</b>
<pre>
mean   = 431.13 μs
stddev = 131.57 μs
min    = 392.36 μs
max    = 2.00 ms
</pre></td></tr><tr><th><h2>sepBy 1000</h2></th></tr><tr><td><b>
runParser sepBy 1000
</b>
<pre>
mean   = 797.28 μs
stddev = 519.89 μs
min    = 461.03 μs
max    = 2.78 ms
</pre></td></tr><tr><td><b>
runParser sepByRec 1000
</b>
<pre>
mean   = 553.79 μs
stddev = 87.00 μs
min    = 467.92 μs
max    = 770.49 μs
</pre></td></tr><tr><th><h2>sepBy 10000</h2></th></tr><tr><td><b>
runParser sepByRec 10000
</b>
<pre>
mean   = 5.76 ms
stddev = 453.62 μs
min    = 5.02 ms
max    = 6.86 ms
</pre></td></tr><tr><th><h2>chainl 10000</h2></th></tr><tr><td><b>
runParser chainl 10000
</b>
<pre>
mean   = 11.26 ms
stddev = 4.65 ms
min    = 6.96 ms
max    = 38.11 ms
</pre></td></tr><tr><td><b>
runParser chainlRec 10000
</b>
<pre>
mean   = 4.57 ms
stddev = 583.18 μs
min    = 4.26 ms
max    = 8.29 ms
</pre></td></tr><tr><th><h2>chainr 1000</h2></th></tr><tr><td><b>
runParser chainr 1000
</b>
<pre>
mean   = 1.28 ms
stddev = 280.53 μs
min    = 818.59 μs
max    = 1.58 ms
</pre></td></tr><tr><td><b>
runParser chainrRec 1000
</b>
<pre>
mean   = 8.18 ms
stddev = 2.15 ms
min    = 5.90 ms
max    = 10.36 ms
</pre></td></tr><tr><th><h2>chainr 10000</h2></th></tr><tr><td><b>
runParser chainrRec 10000
</b>
<pre>
mean   = 609.08 ms
stddev = 9.16 ms
min    = 597.00 ms
max    = 619.30 ms
</pre></td></tr><tr><th><h2>manyTill 1000</h2></th></tr><tr><td><b>
runParser manyTill 1000
</b>
<pre>
mean   = 1.03 ms
stddev = 521.71 μs
min    = 747.51 μs
max    = 3.71 ms
</pre></td></tr><tr><td><b>
runParser manyTillRec 1000
</b>
<pre>
mean   = 841.45 μs
stddev = 211.84 μs
min    = 702.49 μs
max    = 1.70 ms
</pre></td></tr><tr><td><b>
runParser manyTill_ 1000
</b>
<pre>
mean   = 870.80 μs
stddev = 153.41 μs
min    = 770.67 μs
max    = 1.32 ms
</pre></td></tr><tr><td><b>
runParser manyTillRec_ 1000
</b>
<pre>
mean   = 796.78 μs
stddev = 164.55 μs
min    = 702.64 μs
max    = 1.63 ms
</pre></td></tr><tr><th><h2>manyTill 10000</h2></th></tr><tr><td><b>
runParser manyTillRec 10000
</b>
<pre>
mean   = 7.93 ms
stddev = 225.37 μs
min    = 7.55 ms
max    = 8.47 ms
</pre></td></tr><tr><td><b>
runParser manyTillRec_ 10000
</b>
<pre>
mean   = 8.76 ms
stddev = 4.82 ms
min    = 7.49 ms
max    = 41.85 ms
</pre></td></tr><tr><th><h2>mediumJson</h2></th></tr><tr><td><b>
runParser json mediumJson
</b>
<pre>
mean   = 3.79 ms
stddev = 1.46 ms
min    = 3.05 ms
max    = 28.03 ms
</pre></td></tr><tr><td><b>
runTrampoline runParser json mediumJson
</b>
<pre>
mean   = 3.71 ms
stddev = 1.46 ms
min    = 2.99 ms
max    = 24.92 ms
</pre></td></tr><tr><td><b>
StringParser.runParser json mediumJson
</b>
<pre>
mean   = 7.21 ms
stddev = 1.18 ms
min    = 6.68 ms
max    = 37.93 ms
</pre></td></tr><tr><th><h2>largeJson</h2></th></tr><tr><td><b>
runParser json largeJson
</b>
<pre>
mean   = 13.42 ms
stddev = 2.67 ms
min    = 12.05 ms
max    = 38.14 ms
</pre></td></tr><tr><td><b>
runTrampoline runParser json largeJson
</b>
<pre>
mean   = 13.98 ms
stddev = 5.69 ms
min    = 12.12 ms
max    = 54.52 ms
</pre></td></tr><tr><td><b>
StringParser.runParser json largeJson
</b>
<pre>
mean   = 25.77 ms
stddev = 1.30 ms
min    = 24.54 ms
max    = 37.13 ms
</pre></td></tr></table>

File: bench/Main.purs [hunk 1]
```diff
@@ -161,38 +160,109 @@ htmlTableWrap caption benchmark = do
...
+    $ \_ -> Regex.match patternSkidoo stringSkidoo_100000
+
+  log "<th><h2>sepBy 1000</h2></th>"
+  htmlTableWrap "runParser sepBy 1000" $ benchWith 50
+    $ \_ -> runParser string23_1000 $ sepBy anyChar (char '3')
+  htmlTableWrap "runParser sepByRec 1000" $ benchWith 50
+    $ \_ -> runParser string23_1000 $ sepByRec anyChar (char '3')
+
+  log "<th><h2>sepBy 10000</h2></th>"
+  -- sepBy not stack-safe
```
Comment (natefaubion) [inline bench/Main.purs:210]: A stack unsafe parser should be a bug, right?

Comment (jamesdbrock) [reply bench/Main.purs:210]: Yes I think so?

Comment (jamesdbrock) [reply bench/Main.purs:210]: I'm going to merge these benchmarks, then let's talk about it.

**Comment by natefaubion** [comment]
It's not clear to me that runTrampoline is useful other than to prove that it has no effect. runTrampoline on a pure parser is a noop.

**Comment by jamesdbrock** [comment]
> It's not clear to me that runTrampoline is useful other than to prove that it has no effect. runTrampoline on a pure parser is a noop.

Yeah, maybe having proven that `runTrampoline` is a no-op, we could remove that benchmark.

**Pull request merged**